c2corg_api/views/document.py

```diff
@@ -300,14 +301,18 @@ def _delete(self, clazz):
         if not self.request.has_permission('moderator'):
             raise HTTPForbidden('No permission to delete the document')
         id = self.request.validated['id']
-
-        # get the current version of the document
         document = self._get_document(clazz, id)
 
         # check associations constraints
         doc_associations.check_before_deleting_document(document)
 
-        # TODO: image => get filenames
+        if document.type == IMAGE_TYPE:
+            filenames = DBSession.query(Image.filename). \
```

Comment:
I think you want to run this query on `ImageArchive`? Because there is only one entry in the `Image` table for a document id.